test/tests/functional/pbs_cray_smoketest.py

```diff
@@ -0,0 +1,186 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2018 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+from ptl.utils.pbs_crayutils import CrayUtils
+import os
+
+
+@tags('cray', 'smoke')
+class TestCraySmokeTest(TestFunctional):
+
+    """
+    Set of tests that qualifies as smoketest for Cray platform
+    """
+
+    def setUp(self):
+        if not self.du.get_platform().startswith('cray'):
+            self.skipTest("Test suite only meant to run on a Cray")
+        TestFunctional.setUp(self)
+
+    @tags('cray', 'smoke')
+    def test_cray_no_queued_job(self):
```

Comment:
This seems like something you'd want to check in test_cray_login_job or test_cray_compute_job...
I don't see how it is useful as a separate test.  Especially since it is not guaranteed this will run before test_cray_login_job/test_cray_compute_job

Reply:
I have made this part of test_cray_login_job. All jobs are deleted as part of setUp so it does not matter if any previous test that submitted jobs ran before this test.